src/node_env_var.cc

```diff
@@ -198,16 +192,183 @@ static void EnvEnumerator(const PropertyCallbackInfo<Array>& info) {
     if (rc.IsEmpty()) {
       isolate->ThrowException(ERR_STRING_TOO_LONG(isolate));
       FreeEnvironmentStringsW(environment);
-      return;
+      return Local<Array>();
     }
     env_v.push_back(rc.ToLocalChecked());
     p = s + wcslen(s) + 1;
   }
   FreeEnvironmentStringsW(environment);
 #endif
 
-  envarr = Array::New(isolate, env_v.data(), env_v.size());
-  info.GetReturnValue().Set(envarr);
+  return Array::New(isolate, env_v.data(), env_v.size());
+}
+
+std::shared_ptr<KVStore> KVStore::Clone(v8::Isolate* isolate) const {
+  HandleScope handle_scope(isolate);
+  Local<Context> context = isolate->GetCurrentContext();
+
+  std::shared_ptr<KVStore> copy = KVStore::CreateMapKVStore();
+  Local<Array> keys = Enumerate(isolate);
+  uint32_t keys_length = keys->Length();
+  for (uint32_t i = 0; i < keys_length; i++) {
+    Local<Value> key = keys->Get(context, i).ToLocalChecked();
+    CHECK(key->IsString());
+    copy->Set(isolate, key.As<String>(), Get(isolate, key.As<String>()));
+  }
+  return copy;
+}
+
+Local<String> MapKVStore::Get(Isolate* isolate, Local<String> key) const {
+  Mutex::ScopedLock lock(mutex_);
+  String::Utf8Value str(isolate, key);
```

Comment:
Why not use `node::Utf8Value` here? (I believe the only difference is that we use `MaybeStackBuffer` for those?)

Reply:
Yeah, this was copied from the original pieces (I think?) but you’re right, using `node::Utf8Value` is probably the better choice here. Done!